### ISSUE TITLE
Production build fails - ERROR in Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js)

### ISSUE THREAD

**Issue by mishellnovosel** [angular/angular-cli#12604]
### Bug Report or Feature Request (mark with an `x`)
```
- [ X] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ X] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

Node: v10.9.0
Npm: V6.4.1

Package                                                                Version
-----------------------------------------------------------
@angular-devkit/architect                                   0.8.3
@angular-devkit/build-angular                           0.8.5
@angular-devkit/build-optimizer                        0.8.5
@angular-devkit/build-webpack                         0.8.5
@angular-devkit/core                                          0.8.3
@angular-devkit/schematics                              0.8.3
@angular/cdk                                                       6.3.3
@angular/cli                                                          6.2.3
@angular/flex-layout                                            6.0.0-beta.15
@angular/material                                                6.3.3
@ngtools/webpack                                              6.2.5
@schematics/angular                                          0.8.3
@schematics/update                                           0.8.3
rxjs                                                                         6.1.0
typescript                                                              2.7.2
webpack                                                                4.20.2

### Repro steps

Running : "ng build --prod" with following setup:
"fileReplacements": [
  {
     "replace": "src/environments/environment.ts",
     "with": "src/environments/environment.prod.ts"
   }
 ],
 "optimization": true,
 "outputHashing": "all",
 "sourceMap": true,
 "extractCss": true,
 "namedChunks": false,
 "aot": true,
 "extractLicenses": true,
 "vendorChunk": false,
  "buildOptimizer": true

### The log given by the failure

![screen shot 2018-10-16 at 07 28 54](https://user-images.githubusercontent.com/35796562/46994688-c0904980-d115-11e8-9064-76b69956bf64.png)

```
ERROR in Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
TypeError: Cannot read property 'replace' of undefined
    at normalizeBackSlashDirection (/Users/mishellnovosel/Documents/Repositories/cc.angular.stroke/node_modules/webpack/lib/RequestShortener.js:16:17)
    at new RequestShortener (/Users/mishellnovosel/Documents/Repositories/cc.angular.stroke/node_modules/webpack/lib/RequestShortener.js:26:15)
    at new Compiler (/Users/mishellnovosel/Documents/Repositories/cc.angular.stroke/node_modules/webpack/lib/Compiler.js:183:27)
    at Compiler.createChildCompiler (/Users/mishellnovosel/Documents/Repositories/cc.angular.stroke/node_modules/webpack/lib/Compiler.js:425:25)
    at Compilation.createChildCompiler (/Users/mishellnovosel/Documents/Repositories/cc.angular.stroke/node_modules/webpack/lib/Compilation.js:2434:24)
    at Object.pitch (/Users/mishellnovosel/Documents/Repositories/cc.angular.stroke/node_modules/mini-css-extract-plugin/dist/loader.js:70:43)
```

### Desired functionality

I'm expecting for build to pass without an error.

### Mention any other details that might be useful

If the AoT and buildOptimizer are set to false, then the build passes correctly.

**Comment by alan-agius4** [comment]
This seems like a bug but we'll need to look at a reproduction to find and fix the problem. Can you setup a minimal repro please?

You can read [here](https://github.com/angular/angular-cli/blob/master/CONTRIBUTING.md#-submitting-an-issue) why this is needed. A good way to make a minimal repro is to create a new app via `ng new repro-app` and adding the minimum possible code to show the problem. Then you can push this repository to github and link it here.

This might be related to your directory structure so its really important to get an accurate repro to diagnose this.

**Comment by mishellnovosel** [comment]
Hi @alan-agius4,

I've started to create a minimal repo ( which I haven't completed ) and have actually identified the bug. 
There is multiple issues that have happend here:
1. There were some errors in the code that wasn't visible in console or on Production build ( at the time ) because of the "@swimlane/ngx-dnd": "3.2.0"
2. because of the "@swimlane/ngx-dnd": "3.2.0" package, errors weren't displayed as they were, but instead as the "Module build failed" stated in my question
3. After update of swimlane to "@swimlane/ngx-dnd": "5.1.0", the "Module build failed" was gone, and real errors were displayed during the ng build. 
4. After errors fixed, Prod build went through.

### Additional info:
After all code errors were fixed, I've reverted Swimlane to version 3.2.0 just to check if it will work now, and the answer is Yes. Once the code errors were fixed, the 3.2.0 Swimlane passes correctly through the build.

**Comment by alan-agius4** [comment]
Looks like it was due to how the library was packaged.
